src/snowflake/connector/file_transfer_agent_async.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+from .file_transfer_agent import *
+from .file_transfer_agent import _chunk_size_calculator
+from .network_async import EventLoopThreadRunner
+
+# YICHUAN: SnowflakeFileTransferAgentAsync is identical to SnowflakeFileTransferAgent, except for two differences; it
+# owns an EventLoopThreadRunner and uses instances of SnowflakeStorageClientAsync
+
+
+class SnowflakeFileTransferAgentAsync(SnowflakeFileTransferAgent):
+    def __init__(self, *args, **kwargs) -> None:
+        # YICHUAN: This EventLoopThreadRunner may never be used if there is one available in the SnowflakeRestfulAsync
+        # instance owned by SnowflakeConnector, but a thread running an event loop that does nothing is lightweight
+        # and saves us headaches if no SnowflakeConnector instance is associated to a transfer
+        self._loop_runner = EventLoopThreadRunner()
+        self._loop_runner.start()
+
+        super().__init__(*args, **kwargs)
```

Comment:
Usually `super()` calls should be the first thing in methods. You should put this line first in this method and add  everything else afterwards